### PR TITLE
Feat: CI/CD AWS

### PR DESCRIPTION
**Pull request by LazaroQueiroz**
This PR adds a new CI/CD to the 2 AWS EC2 instances.
It automates the build and deploy in those instances, more specifically the deploy for production. 

# How to test it
This feat can be tested when a new tag is added to the repository, then it triggers GitHub Action which creates jobs and the automation starts.

## Changes
The changes were made in: .github/workflows/deploy-oca-yml

### CODE REVIEW

File: .github/workflows/deploy-oca.yml [hunk 1]
```diff
@@ -47,13 +47,72 @@ jobs:
...
+      - name: Pull the latest image
+        run: sudo docker pull ${{ vars.DOCKER_IMAGE_PROD }}:latest
+
       - name: Stop existing container
-        run: sudo docker rm -f ${{ vars.CONTAINER_NAME_PROD }} || true 
-         
+        run: sudo docker rm -f ${{ vars.CONTAINER_NAME_PROD }} || true
+
       - name: Run new container
         run: sudo docker run --name ${{ vars.CONTAINER_NAME_PROD }} -p ${{ vars.HOST_PORT_PROD }}:${{ vars.CONTAINER_PORT }} -d ${{ vars.DOCKER_IMAGE_PROD }}:latest
```
Comment (cilasmarques) [inline .github/workflows/deploy-oca.yml]:
```suggestion
      - name: Run new container
        run: sudo docker run --restart unless-stopped --name ${{ vars.CONTAINER_NAME_PROD }} -p ${{ vars.HOST_PORT_PROD }}:${{ vars.CONTAINER_PORT }} -d ${{ vars.DOCKER_IMAGE_PROD }}:latest
```

File: .github/workflows/deploy-oca.yml [hunk 2]
```diff
@@ -5,12 +5,12 @@ on:
     tags: ["*"]
 
 jobs:
-  build:
-    runs-on: oca-self-hosted-runner
+  build-primaria:
+    runs-on: x1-aws-self-hosted-runner
```
Comment (cilasmarques) [inline .github/workflows/deploy-oca.yml]:
```suggestion
  primary-build:
    runs-on: primary-self-hosted-runner
```

File: .github/workflows/deploy-oca.yml [hunk 3]
```diff
@@ -47,13 +47,72 @@ jobs:
...
+        run: sudo docker run --name ${{ vars.CONTAINER_NAME_PROD }} -p ${{ vars.HOST_PORT_PROD }}:${{ vars.CONTAINER_PORT }} -d ${{ vars.DOCKER_IMAGE_PROD }}:latest
+
+      - name: Add network
+        run: docker network connect oca-network oca-frontend-app
+
+      - name: Prune unused Docker images
+        run: sudo docker system prune -f
+
+  build-failover:
+    runs-on: x2-aws-self-hosted-runner
```
Comment (cilasmarques) [inline .github/workflows/deploy-oca.yml]:
```suggestion
  secondary-build:
    runs-on: secondary-self-hosted-runner
```

File: .github/workflows/deploy-oca.yml [hunk 4]
```diff
@@ -47,13 +47,72 @@ jobs:
...
       - name: Pull the latest image
-        run: sudo docker pull ${{ vars.DOCKER_IMAGE_PROD }}:latest    
-      
+        run: sudo docker pull ${{ vars.DOCKER_IMAGE_PROD }}:latest
+
+      - name: Stop existing container
+        run: sudo docker rm -f ${{ vars.CONTAINER_NAME_PROD }} || true
+
+      - name: Run new container
+        run: sudo docker run --name ${{ vars.CONTAINER_NAME_PROD }} -p ${{ vars.HOST_PORT_PROD }}:${{ vars.CONTAINER_PORT }} -d ${{ vars.DOCKER_IMAGE_PROD }}:latest
```
Comment (cilasmarques) [inline .github/workflows/deploy-oca.yml]:
```suggestion
        run: sudo docker run --restart unless-stopped --name ${{ vars.CONTAINER_NAME_PROD }} -p ${{ vars.HOST_PORT_PROD }}:${{ vars.CONTAINER_PORT }} -d ${{ vars.DOCKER_IMAGE_PROD }}:latest
```

File: .github/workflows/deploy-oca.yml [hunk 5]
```diff
@@ -47,13 +47,72 @@ jobs:
...
+
       - name: Stop existing container
-        run: sudo docker rm -f ${{ vars.CONTAINER_NAME_PROD }} || true 
-         
+        run: sudo docker rm -f ${{ vars.CONTAINER_NAME_PROD }} || true
+
       - name: Run new container
         run: sudo docker run --name ${{ vars.CONTAINER_NAME_PROD }} -p ${{ vars.HOST_PORT_PROD }}:${{ vars.CONTAINER_PORT }} -d ${{ vars.DOCKER_IMAGE_PROD }}:latest
 
+      - name: Add network
```
Comment (cilasmarques) [inline .github/workflows/deploy-oca.yml]:
```suggestion
      - name: Connect to oca network
```

File: .github/workflows/deploy-oca.yml [hunk 6]
```diff
@@ -47,13 +47,72 @@ jobs:
...
-      
+        run: sudo docker pull ${{ vars.DOCKER_IMAGE_PROD }}:latest
+
+      - name: Stop existing container
+        run: sudo docker rm -f ${{ vars.CONTAINER_NAME_PROD }} || true
+
+      - name: Run new container
+        run: sudo docker run --name ${{ vars.CONTAINER_NAME_PROD }} -p ${{ vars.HOST_PORT_PROD }}:${{ vars.CONTAINER_PORT }} -d ${{ vars.DOCKER_IMAGE_PROD }}:latest
+
+      - name: Add network
```
Comment (cilasmarques) [inline .github/workflows/deploy-oca.yml]:
```suggestion
      - name: Connect to oca-network
```

**Review by cilasmarques — CHANGES_REQUESTED** [review]

**Review by cilasmarques — APPROVED** [review]

**Review by RodrigoEC — APPROVED** [review]

**Review by abraaovalentim — APPROVED** [review]
show of balls!

**Pull request merged**